doc/source/language/variables.rst

```diff
@@ -113,3 +205,207 @@ Sets a variable to ``FALSE``. This is useful for the ``RCS`` and ``SAS`` binding
+    variable.  Wouldn't it be easier just to make all variables
+    global?  The answer is twofold: (1) Once a program becomes large
+    enough, trying to remember the name of every variable in the
+    program, and having to keep coming up with new names for new
+    variables, can be a large unmanagable chore, especially with
+    programs written by more than one person collaborating together.
+    (2) Even if you can keep track of all that in your head, there's
+    a certain programming technique known as recursion (TODO - wiki
+    link) in which you actually NEED to have local variable scope for
+    the technique to even work at all.
+
+Examples::
+
+    DECLARE x TO 10. // X is now a global variable with value 10.
+    SET y TO 20. // Y is now a global variable (implicitly) with value 20.
+    DECLARE z TO 0. // Z is now a global variable.
+
+    SET sum to -1. // sum is now an implicitly made global variable, containing -1.
+
+    // A function to return the mean average of all the items in the list
+    // passed into it, under the assumption all the items in the list are
+    // numbers of some sort:
+    DECLARE FUNCTION calcAverage {
+      DECLARE PARAMETER inputList.
+      
+      DECLARE sum TO 0. // sum is now local to this function's body.
+      FOR val IN inputList {
+        SET sum TO sum + val.
+      }.
+      print "Inside calcAverage, sum is " + sum.
+      RETURN sum / inputList:LENGTH.
+    }.
+
+    SET testList TO LIST();
+    testList:ADD(5).
+    testList:ADD(10).
+    testList:ADD(15).
+    print "average is " + calcAverage(testList).
+    print "but out here where it's global, sum is still " + sum.
+
+This example will print::
+
+    Inside calcAverage, sum is 30
+    average is 10
+    but out here where it's global, sum is still -1
+    
+Thus proving that the variable called SUM inside the function is NOT the
+same variable as the one called SUM out in the global main code.
+
+Nesting:
+  The scoping rules are nested as well.  If you attempt to use a
+  variable that doesn't exist in the local scope, the next scope "outside"
+  it wil be used, and if it doesn't exist there, the next scope "outside"
+  that will be used and so on, all the way up to the global scope.  Only
+  if the variable isn't found at the global scope either will it be 
+  implicitly created.
+
+.. _trigger_scope:
+
+Scoping and Triggers:
+:::::::::::::::::::::
+
+Triggers such as:
+
+  - WHEN <expression> { <statements> }.
+
+and
+
+  - ON <boolean variable> { <statements> }.
+
+Do not work predictably when you use local variables in the <expression>
+part of them.  They need to be designed to use global variables only,
+because they outlive the duration of any particular scoping braces.
+You can declare local variables within their <statements> in their bodies,
+just don't use local variables in the trigger conditions.
+
+.. _nolazyglobal:
+
+``NOLAZYGLOBAL``
+::::::::::::::::
```

Comment:
Whoops.  This syntax changed and the docs didn't get updated.  It's now:
```@LAZYGLOBAL OFF.```, and it must be at the top, and ignores scope.  This section needs to be re-written.